### ISSUE TITLE
Built-in container images

### ISSUE THREAD

**Issue by qbx2** [bottlerocket-os/bottlerocket#2974]
**What I'd like:**
I'd like to include large container images (it could be 10GB+) into the AMI image. I am using warm pool to reduce time to pull, but still it costs traffic. It'd be best to include them in AMIs.

**Any alternatives you've considered:**
I have modified the source code for buildsys to be able to `docker-pull` and `docker-save` at build time, and added a package refers to `[[package.metadata.build-package.external-container-images]]`. Then, the package.spec copies the tar file and a service spec to execute `ctr -n k8s.io images import /path/to/tarfile` on boot. If wanted, I have willing to submit a PR.

**Comment by stmcginnis** [comment]
Hey @qbx2, thanks for raising this. I know similar ideas have come up a few times, but I haven't seen a good answer yet.

The idea is really interesting, but the idea and your example solution require creating a custom build of Bottlerocket. Is the idea your proposing that this capability is added to the "core" Bottlerocket to make it easier to make these customized builds?

#2669 is looking at making it easier to create customized builds outside of the main variants included in the Bottlerocket repo. I wonder if this would be a good addition to that effort.

**Comment by qbx2** [comment]
@stmcginnis Yes, the idea is to add the capability to the "core". Also, I expect to leverage this capability easily with out-of-tree Bottlerocket variants.
